app/helpers/task_view_helper.rb

```diff
@@ -1,18 +1,43 @@
 module TaskViewHelper
   def task_view_header(choice)
+    yield (case choice&.task_view_group
```

Comment:
Why use `yield` here?

Reply:
I used yield so it would be consistent with the `take_view_subheader` method.

Reply:
If you want to use it, the approach I'd take would be to remove `display_header?` from the view.

```erb
<% task_view_header(@application_choices, choice) do |heading| %>
```

and

```ruby
def task_view_header(application_choices, choice)
  return unless display_header?(application_choices, choice)
```

Reply:
`display_header?` currently gets called just once per choice. If it were inside the `task_view_header` it would either need to be called again for `task_view_subheader` or `task_view_subheader` would be part of the `task_view_header`. I'm not a fan of either.

I also tried my best not to return html from those methods, that's another reason the method yields just the text content. If the text content is not yielded (only happens for `_subheader` then the markup doesn't get generated either.

I'd like to leave it as it is, but thanks for the suggestions.

Reply:
Well yeah, you'd have to move `task_view_subheader` inside `task_view_header` in the view. But not sure I see an issue with that. 🤷